### PR TITLE
prevent-double-slash-in-github-url

### PR DESCRIPTION
**Pull request by sbeckeriv**
Dearest Reviewer,

The internet is to picky when it comes to locations..

Changes
Consolidate url generation in github
I did not use URI.join like I wanted to because it does not work
like you want it to. Splits and joins seemed the best bet.

Becker

### CODE REVIEW

File: lib/services/github_issues.rb [hunk 1]
```diff
@@ -342,14 +342,27 @@ def server_display_url
...
   end
 
+  def github_url(paths, query = {})
+    paths = [ data.repository.split("/"), *paths.map{|path| path.split("/")}].flatten
+    paths.reject!(&:empty?)
+    server_slash = server_display_url =~ /\/\z/ ? "" : "/"
+    url_string = server_display_url + server_slash + paths.join("/")
+    url = URI.parse(url_string)
+    if query.present?
+      url.query= URI.encode_www_form(query)
```
Comment (stevenpetryk) [inline lib/services/github_issues.rb]: Could you add a space here?

File: lib/services/github_issues.rb [hunk 2]
```diff
@@ -342,14 +342,27 @@ def server_display_url
     end
   end
 
+  def github_url(paths, query = {})
+    paths = [ data.repository.split("/"), *paths.map{|path| path.split("/")}].flatten
```
Comment (stevenpetryk) [inline lib/services/github_issues.rb]: Could you make the spacing consistent around this array?

File: spec/services/github_issues_spec.rb [hunk 3]
```diff
@@ -120,6 +120,20 @@
     end
   end
 
+  describe "#github_url" do
+    it "builts url" do
```
Comment (stevenpetryk) [inline spec/services/github_issues_spec.rb]: *builds

**Comment by sbeckeriv** [comment]
sorry i missed the comments. all fixed.

**Pull request merged**